packages/plugin-hardhat/src/utils/deploy-impl.ts

```diff
@@ -60,7 +60,7 @@ async function getDeployData(
 export async function deployProxyImpl(
   hre: HardhatRuntimeEnvironment,
   ImplFactory: ContractFactory,
-  opts: Options,
+  opts: UpgradeProxyOptions,
```

Comment:
Just a note that we shouldn't really use `UpgradeProxyOptions` here since this file is used for deploys and upgrades, and not all of these support the `call` option from `UpgradeProxyOptions`.

Reply:
That was my first instinct as well, but given that the function is used for both deploys and upgrades, doesn't it make sense for it to support all of the options? Noting that the upgrade options are an extension of the deploy ones.